frontend/src/main/scala/bloop/engine/tasks/CompileTask.scala

```diff
@@ -280,6 +280,8 @@ object CompileTask {
           case FinalNormalCompileResult(p, results) =>
             results.fromCompiler match {
               case Compiler.Result.NotOk(_) => List(p)
+              // Consider success with reported fatal warnings as error to simulate -Xfatal-warnings
+              case Compiler.Result.Success(_, _, _, _, _, _, true) => List(p)
```

Comment:
just wondering if something like that wouldn't be more readable? it makes it obvious what happens here (without reading the comment)

```suggestion
              case s@Compiler.Result.Success if s.reportedFatalWarnings => List(p)
```

Reply:
Sounds good! Have no strong opinions here, will incorporate your suggestion myself.